core-common/src/test/java/org/zalando/nakadi/annotations/validation/DataLakeAnnotationsTest.java

```diff
@@ -39,7 +42,8 @@ public void whenRetentionPeriodThenRetentionReasonRequired() {
                 DataLakeAnnotationValidator.RETENTION_PERIOD_ANNOTATION, "1 day"
         );
         final Set<ConstraintViolation<TestClass>> result = validator.validate(new TestClass(annotations));
-        assertFalse("Missing retention reason is treated valid", result.isEmpty());
+        assertTrue(result.stream().anyMatch(r -> r.getMessage().equals("Retention reason is required, when " +
```

Comment:
Ideally, a failing assertion would report the actual vs. expected values, but given the data types here I see it may be not easy to express.

Reply:
I failed to find an assert utility that out of the box could inspect a set and test for a value while giving clear instruction whether something is found or nothing is found at all.